AWSIoT/Internal/AWSIoTMQTTClient.m

```diff
@@ -767,9 +759,15 @@ - (void)openStreams:(id)sender
             self.connectionAgeTimer = nil;
         }
         [self.session close];
-        
-        if ( self.webSocket) {
+
+        if (self.toDecoderStream != nil) {
+            self.toDecoderStream.delegate = nil;
+            [self.toDecoderStream removeFromRunLoop:runLoopForStreamsThread forMode:NSDefaultRunLoopMode];
             [self.toDecoderStream close];
+            self.toDecoderStream = nil;
```

Comment:
It looks like we reuse this in a few places. Maybe factor it into a private `cleanUpToDecoderStream` method or similar?

Reply:
I was thinking to do this, but ultra paranoid about putting anything on the stack of attempting to remove these items immediately.  But on a second opinion, this is probably too paranoid. Will update and submit.